lib/OpenTracing/Implementation/DataDog/HTTPPropagator.pm

```diff
@@ -0,0 +1,198 @@
+package OpenTracing::Implementation::DataDog::HTTPPropagator;
+
+use strict;
+use warnings;
+use Moo;
+use Carp qw[ croak ];
+
+use constant {
+    STYLE_DATADOG   => 'datadog',
+    STYLE_B3_SINGLE => 'b3 single header',
+    STYLE_B3_MULTI  => 'b3multi',
+    STYLE_W3C       => 'tracecontext',
+    STYLE_NONE      => 'none',
+};
+
+my %styles = (
+    STYLE_DATADOG    ,=> 'OpenTracing::Implementation::DataDog::Propagator::DataDog',
+    STYLE_B3_SINGLE  ,=> 'OpenTracing::Implementation::DataDog::Propagator::B3Single',
+    STYLE_B3_MULTI   ,=> 'OpenTracing::Implementation::DataDog::Propagator::B3Multi',
+    STYLE_W3C        ,=> 'OpenTracing::Implementation::DataDog::Propagator::TraceState',
+    STYLE_NONE       ,=> 'OpenTracing::Implementation::DataDog::Propagator::NoOp',
+);
+
+has style_extract => (
+    is      => 'ro',
+    default => sub {
+          $ENV{DD_TRACE_PROPAGATION_STYLE_EXTRACT}
+       // $ENV{DD_TRACE_PROPAGATION_STYLE}
+       // STYLE_DATADOG
+    },
+);
+
+has style_inject => (
+    is      => 'ro',
+    default => sub {
+          $ENV{DD_TRACE_PROPAGATION_STYLE_INJECT}
+       // $ENV{DD_TRACE_PROPAGATION_STYLE}
+       // STYLE_DATADOG
+    },
+);
+
+has extractor => (
+    is      => 'lazy',
+    handles => [qw/extract/],
+    default => sub {
+        my ($self) = @_;
+        $self->make_propagator($self->style_extract);
+    },
+);
+
+has injector => (
+    is      => 'lazy',
+    handles => [qw/inject/],
+    default => sub {
+        my ($self) = @_;
+        $self->make_propagator($self->style_inject);
+    },
+);
+
+sub make_propagator {
+    my ($self, $style) = @_;
+    my @propagators = map { $_->new } map {
+        $styles{$_} // croak "Unsupported propagation style: $_"
+    } split ',', $style;
```

Comment:
please cleanup the string and remove leading and trailing space, which might be a very easy thing to add, see my comment in the tests

Reply:
Done